web/html/src/manager/content-management/project/project.js

```diff
@@ -84,6 +84,7 @@ const Project = (props: Props) => {
         || (project.environments[1] || {}).status === "building" // promoting 1st env to 2nd: we can't build as it would affect this promotion
         || hasErrors;
 
+    const hasChannelsWithUnsyncedPatches = project.softwareSources.filter(s => s.hasUnsyncedPatches).length > 0;
```

Comment:
Do you really need to store it into a `const`? I see it is used only once.

Reply:
It's only used once, but I prefer extracting the condition to a `const` rather than inlining it in the `Build` component. To me it looks more readable. We use a similar approach for `isProjectEdited`.